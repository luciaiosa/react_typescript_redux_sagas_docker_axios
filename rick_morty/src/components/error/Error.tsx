import React, { FunctionComponent } from "react";
import { styles } from "../../styles/DescriptionStyles";

interface ErrorProps {
    title: string;
}

const Error: FunctionComponent<ErrorProps> = (
    props: ErrorProps
): JSX.Element => {
    const classes = styles();

    return (
        <div className={classes.content}>
            <h2>{props.title}</h2>
            {/* <p className={classes.description}>
                Sorry, there was an error loading the data.
            </p> */}
        </div>
    );
};

export default Error;
