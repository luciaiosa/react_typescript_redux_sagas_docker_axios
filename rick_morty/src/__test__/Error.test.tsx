import {shallow} from 'enzyme';
import React from 'react';
import Error from '../components/error/Error';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Error component', () => {
    it('should render without throwing an error', () => {

        const wrapper = shallow(
            <Error title="Test error"/>
        ); 
        expect(wrapper).toMatchSnapshot();
    }); 
});