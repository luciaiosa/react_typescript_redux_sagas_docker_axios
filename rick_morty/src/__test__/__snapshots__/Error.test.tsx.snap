// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component should render without throwing an error 1`] = `
<div
  className="makeStyles-content-3"
>
  <h2>
    Test error
  </h2>
</div>
`;
